kvs-gbif/src/main/java/org/gbif/kvs/species/BackboneMatchByID.java

```diff
@@ -0,0 +1,165 @@
+    this.checklistbankService = checklistbankService;
+    this.prefixMapping = prefixMapping;
+    this.prefixToChecklistKey = prefixToChecklistKey;
+  }
+
+  Integer lookupBackboneKey(Identification identification, Set<OccurrenceIssue> issues) {
+    Integer backboneUsageKey;
+    String scientificName = identification.getScientificName();
+    String scientificNameID = identification.getScientificNameID();
+    String taxonID = identification.getTaxonID();
+    String taxonConceptID = identification.getTaxonConceptID();
+
+    LOG.debug("Attempting to locate a backbone taxon for scientificNameID[{}]", scientificNameID);
+    backboneUsageKey = lookupBackboneKey(scientificNameID, scientificName, issues, TAXON_MATCH_SCIENTIFIC_NAME_ID_IGNORED, SCIENTIFIC_NAME_ID_NOT_FOUND);
+
+    if (backboneUsageKey != null) {
+      LOG.debug("Backbone key[{}] found using scientificNameID[{}]. No further taxon IDs will be used", backboneUsageKey, scientificNameID);
+      issues.add(TAXON_MATCH_TAXON_ID_IGNORED);
+      issues.add(TAXON_MATCH_TAXON_CONCEPT_ID_IGNORED);
+    } else {
+      LOG.debug("Attempting to locate a backbone taxon for taxonID[{}]", taxonID);
+      backboneUsageKey = lookupBackboneKey(taxonID, scientificName, issues, TAXON_MATCH_TAXON_ID_IGNORED, TAXON_ID_NOT_FOUND);
+    }
+
+    if (backboneUsageKey != null) {
+      LOG.debug("Backbone key[{}] found using taxonID[{}]. No further taxon IDs will be used", backboneUsageKey, taxonID);
+      issues.add(TAXON_MATCH_TAXON_CONCEPT_ID_IGNORED);
+    } else {
+      LOG.debug("Attempting to locate a backbone taxon for taxonConceptID[{}]", taxonConceptID);
+      backboneUsageKey = lookupBackboneKey(taxonConceptID, scientificName, issues, TAXON_MATCH_TAXON_CONCEPT_ID_IGNORED, TAXON_CONCEPT_ID_NOT_FOUND);
+    }
+
+    LOG.debug("Backbone key was not found using any of the name or taxa IDs");
+    return backboneUsageKey;
+  }
+
+  Integer lookupBackboneKey(String id, String name, Set<OccurrenceIssue> issues, OccurrenceIssue ignoredFlag, OccurrenceIssue notFoundFlag) {
+    Integer backboneUsageKey = null;
+    if (id != null) {
+      LOG.debug("Attempting to locate a backbone taxon for ID[{}]", id);
+
+      String adjustedID = prefixReplaceId(id, prefixMapping).trim();
+      String checklistKey = checklistKeyForID(adjustedID, prefixToChecklistKey);
+      if (checklistKey != null && adjustedID.length()>0) {
+
+        // Short circuit: if the checklist is already pointing to the backbone we don't need any further lookup
+        if (checklistKey.equalsIgnoreCase(NUB_DATASET_KEY.toString())) {
+          try {
+            backboneUsageKey = Integer.parseInt(adjustedID);
+          } catch (NumberFormatException e) {
+            issues.add(notFoundFlag);
+            LOG.debug("Adjusted ID[{}] is no backbone key integer", adjustedID);
+          }
+
+        } else {
```

Comment:
when GBIF ids are found no flags are raised

Reply:
Good point, I'll add

Reply:
Correction: I think it's working correctly with regard to this.
If a backbone key is found in scientificNameID, then the logical will add ignored flags on the taxonID and taxonConceptID